### PR TITLE
Fix issue with nested operators, add ability to filter whereNull on fields

### PR DESCRIPTION
**Pull request by craicoverflow**
## Tasks

- There was an issue when nested and/or/not operators - it only worked one level deep - this fixes that.
- Add ability to perform filtering by checking if a field is null/non-null
- This also adds ability to filter between ranges.

## Verification

To filter results on a null field:

```graphql
query findNotes {
  findNotes(filter: {items:{eq:null}}) {
    items {
      id
      title
      items
    }
  }
}
```

To perform nested operator filtering (or -> not):

```graphql
query findNotes {
  findNotes(filter: {items:{eq:null}, or:{not:{id:{eq:null}}}}) {
    items {
      id
      title
      items
    }
  }
}
```

### CODE REVIEW

**Comment by craicoverflow** [comment]
@wtrocki here is my PR - I have not rebased like you asked.

**Comment by wtrocki** [comment]
Working on rebase. I will also merge that and let you review things in case rebase went bad. 

Please keep local copy of the branch in case things will be lost

File: packages/graphback-codegen-schema/src/definitions/schemaDefinitions.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
+import { GraphQLInputObjectType, GraphQLFloat, GraphQLList, GraphQLBoolean, GraphQLInt, GraphQLString, GraphQLID, GraphQLEnumType, GraphQLObjectType, GraphQLNonNull, getNullableType, GraphQLField, getNamedType, isInputType, GraphQLInputField, GraphQLFieldMap, isScalarType, GraphQLInputFieldMap, GraphQLScalarType, GraphQLInputFieldConfig, GraphQLInputType, GraphQLOutputType, GraphQLNamedType } from "graphql";
```
Comment (wtrocki) [inline packages/graphback-codegen-schema/src/definitions/schemaDefinitions.ts:1]: I have tons of conflicts on this file that I cannot resolve easily without context

**Comment by wtrocki** [comment]
I have tried to do `git rebase` but this gave be like 70 conflicts as this PR contains tons of small commits. 

`git merge origin/crudspec-feature` gives less but they are hard to resolve. 
Can you help me with resolving conflicts.

**Comment by ssd71** [comment]
Hi, I tried out this query:
```
{
   "filter":{
      "not":{
         "or":[
            {
               "title":{
                  "contains":"pea"
               }
            },
            {
               "title":{
                  "contains":"love"
               }
            }
         ]
      }
   }
}
```

But I got this result:
```
{
  "data": {
    "findNotes": {
      "items": [
        {
          "title": "pea",
          "description": "pea354231ce"
        },
        {
          "title": "pea12312414",
          "description": "pea354love231ce"
        },
        {
          "title": "love12312414",
          "description": "pea354love231ce"
        },
        {
          "title": "love12312414",
          "description": "reaf3112"
        }
      ]
    }
  }
}
````
I was expecting none of the items would match the or block, feel free to point out if I made a mistake somewhere :P

**Comment by craicoverflow** [comment]
> I have tried to do `git rebase` but this gave be like 70 conflicts as this PR contains tons of small commits.
> 
> `git merge origin/crudspec-feature` gives less but they are hard to resolve.
> Can you help me with resolving conflicts.

Maybe I should squash into 2-3 commits?

**Comment by wtrocki** [comment]
> Maybe I should squash into 2-3 commits? 

THat will do the job as well

**Comment by craicoverflow** [comment]
Moved to https://github.com/aerogear/graphback/pull/1254

**Comment by craicoverflow** [comment]
@ssd71 

You filter outputs the following SQL:

To expect an empty list you should use AND instead of OR. With OR, `not contains pea` returns 2 items and `not contains love` returns 2 items, so you get 4 in total.

SQL output:

`select * from "note" where not "title" like '%pea%' or not "title" like '%love%'`

However `not "title" like '%pea%'` should really be `"title" not like '%pea%'` so I created https://github.com/aerogear/graphback/issues/1256 so that we can properly map this operator

**Pull request closed without merging**